src/measure_theory/integral/lebesgue.lean

```diff
@@ -798,6 +801,22 @@ eq_top_iff.mpr $
 calc ∞ = ∞ * μ {x | ∞ ≤ f x} : by simp [mul_eq_top, hμf.ne.symm]
    ... ≤ ∫⁻ x, f x ∂μ : mul_meas_ge_le_lintegral₀ hf ∞
 
+lemma set_lintegral_eq_top_of_measure_eq_top_pos (hf : ae_measurable f (μ.restrict s))
+  (hs : null_measurable_set s μ) (hμf : 0 < μ {x ∈ s | f x = ⊤}) :
```

Comment:
```suggestion
  (hs : null_measurable_set s μ) (hμf : 0 < μ {x ∈ s | f x = ∞}) :
```
Same thing several times below.

Reply:
Also, your assumption hμf should instead be that the measure is nonzero. And you shouldn't need `hs`.